src/v/cluster/rm_stm.h

```diff
@@ -199,13 +267,13 @@ class rm_stm final : public raft::persisted_stm<> {
     ss::future<result<kafka_result>> transactional_replicate(
       model::batch_identity, model::record_batch_reader);
 
-    ss::future<result<kafka_result>> do_sync_and_transactional_replicate(
+    ss::future<result<kafka_result>> do_transactional_replicate(
+      model::term_id,
       tx::producer_ptr,
       model::batch_identity,
-      model::record_batch_reader,
-      ssx::semaphore_units);
+      model::record_batch_reader);
 
-    ss::future<result<kafka_result>> do_transactional_replicate(
+    ss::future<result<kafka_result>> do_transactional_replicate0(
```

Comment:
nit: this name is confusing, why `0` is appended at the end ?

Reply:
oh we have two flavors of `do_transactional_replicate` ..  one dispatches the work and does high level checks, the other (with `0`) does the meat of the work.. I thought this is a pretty common pattern .. send() and send0() etc.. I can rename it.

Reply:
I thought the pattern was to prepend "do_", but here we already have it :)